### PR TITLE
Add ability for networking stub to handle multiple requests

### PR DESCRIPTION
**Pull request by aliak00**

### CODE REVIEW

File: Example/Tests/AutoRefreshURLProtocolTests.swift [hunk 1]
```diff
@@ -90,12 +90,21 @@ class AutoRefreshURLProtocolTests: QuickSpec {
             let refreshUrl = "/oauth/token"
             let successData = "i am not google"
 
-            self.stub(uri(refreshUrl), try! Builders.load(file: "valid-refresh", status: 200))
-
-            self.stub(uri(wantedUrl), Builders.sequentialBuilder([
-                try! Builders.load(file: "empty", status: 401),
-                Builders.load(string: successData, status: 200),
-            ]))
+            let testingNetworkingProxy = TestingNetworkingProxy()
```
Comment (Torsph) [inline Example/Tests/AutoRefreshURLProtocolTests.swift]: Do you need this code when you are setting it before each test in the Example/Tests/Configuration.swift ?

File: SchibstedAccount/Core/Networking/StubbedNetworkingProxy.swift [hunk 2]
```diff
@@ -223,32 +297,44 @@ class StubbedNetworkingProxy: NetworkingProxy {
...
         }
 
+        // Check if there're exact matches on this URL. If there are, first check if we are allowed
+        // to proceed with it (defaults to true) and then return a mock data task
         for stub in type(of: self).urls[url] ?? [] {
             if stub.proceed(with: request) {
+                // Incase we are an arrayOfData, we need to remove the first elemenet since it is now "used up"
+                defer {
+                    if case var .arrayOfData(datas)? = stub.responseData {
+                        datas.remove(at: 0)
```
Comment (Torsph) [inline SchibstedAccount/Core/Networking/StubbedNetworkingProxy.swift]: Should this be extracted into its own functions?

File: Example/Tests/Configuration.swift [hunk 3]
```diff
@@ -24,7 +24,15 @@ class SchibstedAccountConfiguration: QuickConfiguration {
...
             Utils.cleanupKeychain()
             PasswordlessTokenStore.clear()
-            Networking.proxy = TestingNetworkingProxy()
+
+            //
+            // Before each test, make sure we reset the proxy to use the testing networking proxy incase a test changes the configuration
+            // And make sure the internal forwarding proxy if the testing proxy is the default one.
+            //
+            let testingNetworkingProxy = TestingNetworkingProxy()
+            testingNetworkingProxy.internalProxy = DefaultNetworkingProxy()
```
Comment (Torsph) [inline Example/Tests/Configuration.swift]: Should this be `StubbedNetworkingProxy`? We changed that earlier to make the stubbing work.

File: Example/Tests/Utils/TestingNetworkingProxy.swift [hunk 4]
```diff
@@ -43,7 +43,7 @@ class TestingNetworkingProxy: NetworkingProxy {
         return self.internalProxy.session
     }
 
-    let internalProxy: NetworkingProxy = DefaultNetworkingProxy()
+    var internalProxy: NetworkingProxy = DefaultNetworkingProxy()
```
Comment (lrossi) [inline Example/Tests/Utils/TestingNetworkingProxy.swift]: Can this be passed in the initializer?

**Pull request merged**